CefSharp.Wpf/Rendering/InteropBitmapInfo.cs

```diff
@@ -31,14 +32,24 @@ public override void ClearBitmap()
 
         public override void Invalidate()
         {
-            Bitmap.Invalidate();
+            if (Bitmap != null)
+            {
+                Bitmap.Invalidate();
+            }
         }
 
         public override BitmapSource CreateBitmap()
         {
             var stride = Width * BytesPerPixel;
 
-            Bitmap = (InteropBitmap)Imaging.CreateBitmapSourceFromMemorySection(FileMappingHandle, Width, Height, PixelFormat, stride, 0);
+            if (FileMappingHandle == IntPtr.Zero)
+            {
+                ClearBitmap();
```

Comment:
I think my update fulfils your request. Let me know.

On 5 May 2015 at 22:54, Alex Maitland <notifications@github.com> wrote:

> In CefSharp.Wpf/Rendering/InteropBitmapInfo.cs
> <https://github.com/cefsharp/CefSharp/pull/1003#discussion_r29717350>:
>
> >          }
> >
> >          public override BitmapSource CreateBitmap()
> >          {
> >              var stride = Width * BytesPerPixel;
> >
> > -            Bitmap = (InteropBitmap)Imaging.CreateBitmapSourceFromMemorySection(FileMappingHandle, Width, Height, PixelFormat, stride, 0);
> > +            if (FileMappingHandle == IntPtr.Zero)
> > +            {
> > +                ClearBitmap();
>
> CreateBitmap() is usually called after the bitmap has already been
> cleared. In the scenario your trying to handle hear is it possible for bitmap
> != null?
>
> Ideally would fail fast, have a code block like this as the first
> operation.
>
> // Unable to create bitmap without valid File Handle (Most likely control is being disposed)if (FileMappingHandle == IntPtr.Zero)
> {
>   return null;
> }
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cefsharp/CefSharp/pull/1003/files#r29717350>.
>